modules/base/src/time/calendar_duration.fz

```diff
@@ -90,7 +90,48 @@ is
   # create a string representation of this duration.
   #
   public redef as_string String =>
-    "NYI"
+    lm : mutate.
+    lm ! ()->
+      is_positive := lm.env.new bool false
+      time_prefix := lm.env.new bool false
+      s := lm.env.new String "P"
+
+      with_time_prefix (x String) =>
+        if !time_prefix.get
+          time_prefix <- true
+          "{s.get}T{x}"
+        else
+          "{s.get}{x}"
+
+      if years > 0
+        s <- "{s.get}{years}Y"
+        is_positive <- true
+
+      if months > 0
+        s <- "{s.get}{months}M"
+        is_positive <- true
+
+      if days > 0
+        s <- "{s.get}{days}D"
+        is_positive <- true
+
```

Comment:
Instead of `with_time_prefix` I would prefer sth. like `needs_time_prefix` then add the prefix here.